x-pack/plugins/stack_connectors/server/connector_types/jira/service.ts

```diff
@@ -435,47 +378,29 @@ export const createExternalService = (
   };
 
   const getFieldsByIssueType = async (issueTypeId: string) => {
-    const capabilitiesResponse = await getCapabilities();
-    const supportsNewAPI = hasSupportForNewAPI(capabilitiesResponse);
     try {
-      if (!supportsNewAPI) {
-        const res = await request({
-          axios: axiosInstance,
-          method: 'get',
-          url: createGetIssueTypeFieldsUrl(getIssueTypeFieldsOldAPIURL, issueTypeId),
-          logger,
-          configurationUtilities,
-          connectorUsageCollector,
-        });
-
-        throwIfResponseIsNotValid({
-          res,
-        });
-
-        const fields = res.data.projects[0]?.issuetypes[0]?.fields || {};
-        return normalizeFields(fields);
-      } else {
-        const res = await request({
-          axios: axiosInstance,
-          method: 'get',
-          url: createGetIssueTypeFieldsUrl(getIssueTypeFieldsUrl, issueTypeId),
-          logger,
-          configurationUtilities,
-        });
-
-        throwIfResponseIsNotValid({
-          res,
-        });
-
-        const fields = res.data.values.reduce(
-          (acc: { [x: string]: {} }, value: { fieldId: string }) => ({
-            ...acc,
-            [value.fieldId]: { ...value },
-          }),
-          {}
-        );
-        return normalizeFields(fields);
-      }
+      const res = await request({
+        axios: axiosInstance,
+        method: 'get',
+        url: createGetIssueTypeFieldsUrl(getIssueTypeFieldsUrl, issueTypeId),
+        logger,
+        configurationUtilities,
+      });
+
+      throwIfResponseIsNotValid({
+        res,
+      });
+
+      // Cloud returns fields and Data Center returns values
+      const { fields: rawFields, values } = res.data;
+      const fields = (rawFields || values).reduce(
```

Comment:
Same as above. Ideally, we would let the user configure which one is used and then we pick the response format based on that setting.